packages/@divvi/mobile/src/divviProtocol/slice.ts

```diff
@@ -0,0 +1,49 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { getRehydratePayload, REHYDRATE, RehydrateAction } from 'src/redux/persist-helper'
+import { RootState } from 'src/redux/store'
+
+interface ReferralState {
+  successfulReferrals: {
+    [key: string]: boolean // key is `${divviId}-${campaignIds.join(',')}` where campaignIds are sorted
+  }
```

Comment:
nit: maybe we could store per campaignId `${divviId}-${campaignId}` to only register new campaigns instead of the whole list everytime, when we update the campaignIds list.